### PR TITLE
feat: rollover merkle trees, refactor: unify init methods

### PR DESCRIPTION
**Pull request by ananas-block**
Changes:
- light-verifier
  - contains onchain zkp inclusion and non-inclusion verification logic so that it can easily be used in other programs
  - includes verifyingkeys
- compressed-pda
  - system program is mandatory now to pay rollover fees
- hashset
  - made parameters public so that I can assert them
- reference indexed Merkle tree
  - added init function
- account compression
  - append_leaves & insert_address
    - added rollover fee and requirement for system program and fee payer
    - the fee payer is not a pda but the signer of the actual transaction
  - initialized address Merkle tree and queue & initialized state Merkle tree and queue
    - unified into one instruction for security (every Merkle tree and queue always have an associated mt / queue)
    - fees are computed in a unified fashion
  - added rollover state and address Merkle tree and queue instructions
  - added properties to State and Address Merkle tree structs:
    1. ``rollover_fee``(is calculated onchain if threshold is set)
    2. ``tip`` (custom tip that is passed with instruction data)
    3.  ``rollover_threshold`` (in basispoints determines to what degree a Mt has to be filled to be eligible for rollover eg 95%)
    4. ``rolledover_slot`` (The slot when account was rolledover to prevent additional rollovers)
    5. ``close_threshold`` (Placeholder for a threshold after how many slots after rollover remaining state can be moved to next tree and this tree and queue can be closed)
  - remove compressedSol omnibus account and init functions
     - these are not necessary anymore because we need this pda to sign for itself and accounts with data cannot sign a cpi

### CODE REVIEW

File: js/stateless.js/src/idls/light_system.ts
Comment (SwenSchaeferjohann) [inline js/stateless.js/src/idls/light_system.ts:1]: file is called `light_system` but exports `lightRegistry`, we should probably adapt the file name.

File: js/stateless.js/src/state/types.ts
Comment (SwenSchaeferjohann) [inline js/stateless.js/src/state/types.ts:1]: it does seem like it's applying changes that you had previously already merged into main

File: merkle-tree/indexed/src/reference.rs
Comment (SwenSchaeferjohann) [inline merkle-tree/indexed/src/reference.rs:62]: would it be sensible to add an inline comment describing `init` here?

File: programs/account-compression/src/instructions/append_leaves.rs
Comment (SwenSchaeferjohann) [inline programs/account-compression/src/instructions/append_leaves.rs:84]: are these leftovers or on purpose

File: programs/account-compression/src/instructions/append_leaves.rs
Comment (SwenSchaeferjohann) [inline programs/account-compression/src/instructions/append_leaves.rs:100]: same question here

File: programs/account-compression/src/instructions/append_leaves.rs
Comment (SwenSchaeferjohann) [inline programs/account-compression/src/instructions/append_leaves.rs:127]: same question here

File: programs/compressed-pda/src/verify_state.rs
Comment (SwenSchaeferjohann) [inline programs/compressed-pda/src/verify_state.rs:156]: leftover comment?

File: programs/compressed-pda/tests/test.rs
Comment (SwenSchaeferjohann) [inline programs/compressed-pda/tests/test.rs:706]: can this be removed ?

File: programs/compressed-token/src/spl_compression.rs
Comment (SwenSchaeferjohann) [inline programs/compressed-token/src/spl_compression.rs:90]: leftover msg?

**Review by SwenSchaeferjohann — APPROVED** [review]
lgtm! 
doing optimistic approval to unblock. pls do check the comments before merging

File: js/compressed-token/src/idl/light_compressed_token.ts
Comment (ananas-block) [reply js/compressed-token/src/idl/light_compressed_token.ts:157]: needs to be writable to transfer sol, rollover fees are transferred from the feepayer

File: programs/compressed-pda/src/verify_state.rs
Comment (ananas-block) [reply programs/compressed-pda/src/verify_state.rs:156]: renenabled heap neutrality

**Pull request merged**
